Vizzuality/heco-invest
# PD users can see a detailed view of the open calls they applied to in the Dashboard

## Description

**In this PR:**  
- `Breadcrumbs` component changes:  
  - Added a `theme` prop
    _And added both `light` and `dark` color themes. The former is used in the dashboard's open call application's details breadcrumb, the latter in the public pages_  
  - Added a prop to hide segments  
    _needed to hide the "Dashboard" segment in the open call application's details breadcrumb, in order to better match the designs_  
- `DashboardLayout` changes:  
  - Added a prop to switch header "types"  
    _`account` to display the standard dashboard header, `breadcrumbs` to display breadcrumbs instead_
- New `WithdrawApplicationModal` component  
  _The PD can withdraw their open call application from both the open call applications listing and details pages, so the  
 confirmation modal logic was extracted into a separate and reusable component_
- New service/hooks to get an account open call application by its id
- New `OpenCallDetailsPage`
  _where a project developer can see a detailed view of the open calls they applied to_


**Other changes:**  
- Minor `Button` component classnames changes
  _In order to address inconsistent button sizing, more noticeable in confirmation modals where the "Cancel" and "Confirm" buttons would have slightly different heights_

## Testing instructions

Verify that a project developer can:  
- see the details of their open call applications  
- withdraw their open call applications from both the dashboard list and the details view  
- the application details view's breadcrumbs display correctly  
- the information in the application details view is correct  

## Tracking

[LET-899 | FE - PD users can see a detailed view of the open calls they applied to in the Dashboard](https://vizzuality.atlassian.net/browse/LET-899)
[LET-982 | FE - PD users can withdraw a project application in their dashboard](https://vizzuality.atlassian.net/browse/LET-982)
